src/prefixTrie.c

```diff
@@ -0,0 +1,156 @@
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
+#define CHILDREN_SIZE 30
+
+// Structure for a trie. Each node stores a char as data and its children as an array.
+typedef struct Trie {
+    char data;
+    struct Trie *children[CHILDREN_SIZE];
+} Trie;
+
+// Default Constructor
+Trie* trieCreate() {
+    return calloc(1, sizeof(Trie));
+}
+
+// Parameterized Constructor
+Trie* trieCreateByData(char c) {
+    Trie* trie = trieCreate();
+    trie->data = c;
+    return trie;
+}
+
+// Insert word into obj
+void trieInsert(Trie* obj, char * word) {
```

Comment:
Why is this insert method so complicated? Is there a simpler way or is this how it has to be?